server/activityItem/model.ts

```diff
@@ -9,34 +9,44 @@ import {
 	Index,
 } from 'sequelize-typescript';
 import type { InferAttributes, InferCreationAttributes, CreationOptional } from 'sequelize';
-import { ActivityItemKind, ActivityItemPayload } from 'types';
+import { InsertableActivityItem, RecursiveAttributes } from 'types';
 
 @Table
-export class ActivityItem extends Model<
-	InferAttributes<ActivityItem>,
-	InferCreationAttributes<ActivityItem>
+// export class ActivityItem<Kind extends ActivityItemKind = ActivityItemKind> extends Model<
+// 	InferAttributes<ActivityItem<Kind>>,
+// 	InferCreationAttributes<ActivityItem<Kind>>
+// > {
+export class ActivityItem<T extends InsertableActivityItem = InsertableActivityItem> extends Model<
+	InferAttributes<ActivityItem<T>, { omit: Extract<keyof ActivityItem<T>, keyof T> }> & T,
+	InferCreationAttributes<ActivityItem<T>, { omit: Extract<keyof ActivityItem<T>, keyof T> }> & T
 > {
+	// this overrides the default Date type to be compatible with existing code
+	declare createdAt: CreationOptional<string>;
+	declare updatedAt: CreationOptional<string>;
```

Comment:
Ive added this to all Models. Maybe cleaner to create a new Model class that already takes care of this.

Ive chosen `string` because that is how the values are used in the code (almost always).
And, because when serialized that is the correct value.

If we left it at the default (`Date`), then `ActivityItem.toJSON()` would still have `updatedAt: Date`. 
Now, we override `toJSON` anyway, so we could also just make it turn `Date`s to `string`s properly, but this seemed simpler. Let me know if you think the other way is better though, I kind of think it is but will run into more edge cases.